src/mpl/src/gpu/mpl_gpu_ze.c

```diff
@@ -821,17 +821,23 @@ static inline int free_ipc_handle_cache(MPL_ze_ipc_handle_entry_t * cache_entry)
         munmapFunction(cache_entry->nfds, cache_entry->mapped_ptr, cache_entry->mapped_size);
         if (!cache_entry->handle_cached) {
             /* free the ipc handle itself */
-            for (int i = 0; i < cache_entry->nfds; i++) {
-                ret = zeMemPutIpcHandle(ze_context, cache_entry->ipc_handle.ipc_handles[i]);
-                ZE_ERR_CHECK(ret);
+            /* FIXME: in implicit scaling mode, lack of L0 function to free two handles */
+            if (cache_entry->nfds == 1) {
```

Comment:
Is this correct? `nfds == 1` would make the following `for-`loop trivial.

Reply:
I cleaned it up a little bit. But it was intended. We can not handle the two handle case right now due to lack of an extension function in level zero.

Reply:
What happens under implicit scaling mode? We simply not free ipc handle cache?

Reply:
You are right, there is a FIXME comment about that. This has been a known issue since previous version (until the new zeMemPutIpcHandle API was introduced to allow a IPC handle to be freed at the sender side). Note that the receiver side has OpenIpcHandle and CloseIpcHandle, so it is free of such issue. This fix, right now is just a workaround to avoid crash. However, we will need a fix in level zero to implement a variant of the PutIpcHandle for implicit scaling mode. 
As a recommendation to the users, the users are advised to turn off implicit scaling.

Reply:
I see. How about:
```       
        if (cache_entry->nfds == 1) {
            ret = zeMemPutIpcHandle(ze_context, cache_entry->ipc_handle.ipc_handles[0]);
            ZE_ERR_CHECK(ret);
        } else {
            /* FIXME: in implicit scaling mode, lack of L0 function to free two handles */
        }
```

This makes it clear that `FIXME` applies to the implicit scaling branch and that branch is currently missing handle cache cleanups.